Fix Logging HttpRequest#method

blowmage opened the pull request:
`HttpRequest#method` is stomping on `Object#method` and this is preventing calls to `Object#method` to be made. This fix is to maintain both behaviors by checking whether an argument was passed to `#method`. The `HttpRequest#method` behavior is used when called with no arguments, and the `Object#method`, behavior is used when called with arguments.

```ruby
require "google/cloud/logging"

logging = Google::Cloud::Logging.new

entry = logging.entry
entry.http_request.method = :GET

# call with HttpRequest#method behavior
entry.http_request.method #=>:GET

# call with Object#method behavior
entry.http_request.method(:validated)
#=> #<Method: Google::Cloud::Logging::Entry::HttpRequest#validated>
```

[fixes #1798]

quartzmo commented:
@blowmage What do you think about @dazuma's [suggestion](https://github.com/GoogleCloudPlatform/google-cloud-ruby/issues/1798#issuecomment-343310109) to rename `method` to `request_method`?

> I'd advocate for renaming the accessor to request_method since that's the name of the underlying proto field anyway.

blowmage commented:
I think that would be a breaking change and I'd prefer not to force a 2.0 version to fix this. Thankfully the Logging `HttpRequest#method` and the `Object#method` signatures are different, so we can support both.

I am not opposed to renaming the method when we do move to 2.0 though.

quartzmo commented:
Not a breaking change: Move docs/recommended usage to `request_method`, but keep support for `method` as you have done here. I think it's a good idea so that users don't feel worried or concerned about the overload of `method`.

blowmage commented:
K, I added `HttpRequest#request_method` and added deprecation notice to the `HttpRequest#method` documentation. Calls to `HttpRequest#method` will continue to work so existing code won't need to be changed.

blowmage commented:
@dazuma I'd really like your approval before accepting this.

dazuma commented on google-cloud-logging/lib/google/cloud/logging/entry/http_request.rb:
```diff
@@ -33,7 +33,30 @@ def initialize
+          #   `"POST"`. (String)
+          def method *args
+            # Call Object#method when args are present.
+            return super if args.any?
```
Should this be `unless args.empty?`?
because `[nil].any? # => false`

blowmage replied:
@dazuma Updated with test coverage.
